xray/plot/plot.py

```diff
@@ -492,13 +367,28 @@ def _plot2d(plotfunc):
     plotfunc.__doc__ = '\n'.join((plotfunc.__doc__, commondoc))
 
     @functools.wraps(plotfunc)
-    def newplotfunc(darray, x=None, y=None, ax=None, xincrease=None, yincrease=None,
-                    add_colorbar=True, add_labels=True, vmin=None, vmax=None, cmap=None,
-                    center=None, robust=False, extend=None, levels=None, colors=None,
-                    **kwargs):
+    def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
+                    col_wrap=None, xincrease=None, yincrease=None,
+                    add_colorbar=True, add_labels=True, vmin=None, vmax=None,
+                    cmap=None, center=None, robust=False, extend=None,
+                    levels=None, colors=None, **kwargs):
         # All 2d plots in xray share this function signature.
         # Method signature below should be consistent.
 
+        # Handle facetgrids first
+        if row or col:
+            allargs = locals().copy()
+            allargs.update(allargs.pop('kwargs'))
+
+            # Allows use of better FacetGrid defaults
+            assert allargs.pop('add_labels')
+            assert allargs.pop('add_colorbar')
+
+            # Need the decorated plotting function
+            allargs['plotfunc'] = globals()[plotfunc.__name__]
+
+            return _easy_facetgrid(**allargs)
```

Comment:
I tried that approach first. But it was the wrong `plotfunc`, since this is done in the scope of the decorating function, and therefore doesn't yet have the desired signature. We need to get the one that's already decorated, and that one is in `globals`.